daemon/src/model.rs

```diff
@@ -647,43 +647,41 @@ impl str::FromStr for FundingRate {
     }
 }
 
-/// Fee paid by the taker to the maker for opening a cfd.
-#[derive(Debug, Copy, Clone, PartialEq, PartialOrd, Serialize, Deserialize)]
+#[derive(thiserror::Error, Debug)]
+pub enum ConversionError {
+    #[error("Underflow")]
+    Underflow,
+    #[error("Overflow")]
+    Overflow,
+}
+
+/// Opening fee is always payed from taker to maker
+#[derive(Copy, Clone, Debug, Serialize, Deserialize, PartialEq)]
 pub struct OpeningFee {
-    amount_sat: u64,
+    #[serde(with = "bdk::bitcoin::util::amount::serde::as_sat")]
+    fee: Amount,
 }
 
 impl OpeningFee {
-    pub fn new(amount: Amount) -> Self {
-        Self {
-            amount_sat: amount.as_sat(),
-        }
+    pub fn new(fee: Amount) -> Self {
+        Self { fee }
     }
 
-    /// Value in satoshis for the taker
-    pub fn to_u64(&self) -> u64 {
-        self.amount_sat
-    }
+    fn into_signed_amount(self, role: Role) -> SignedAmount {
+        let fee: i64 = self.fee.as_sat().try_into().expect("not to overflow");
 
-    pub fn amount_for(&self, role: Role) -> SignedAmount {
-        let amount_sat: i64 = self.amount_sat.try_into().expect("to convert");
-        let amount_sat = match role {
-            Role::Maker => -amount_sat,
-            Role::Taker => amount_sat,
+        let signed = match role {
+            Role::Maker => -fee,
+            Role::Taker => fee,
         };
-        SignedAmount::from_sat(amount_sat)
-    }
-}
 
-impl From<Amount> for OpeningFee {
-    fn from(amount: Amount) -> Self {
-        Self::new(amount)
+        SignedAmount::from_sat(signed)
     }
 }
 
 impl fmt::Display for OpeningFee {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        self.amount_sat.fmt(f)
+        self.fee.as_sat().fmt(f)
```

Comment:
❓ So the intention is to display an absolute value? Just confirming.

Reply:
I think yes, because this is just the internal representation. For the projection we use a signed amount at the moment (for the complete fee).

Reply:
I guess the confusing part is that these `Display` traits in the model are for the sqlx macro to be able to store it as a string (and `FromStr` to parse it back) - it took me a while to realise this!

Reply:
> I guess the confusing part is that these `Display` traits in the model are for the sqlx macro to be able to store it as a string (and `FromStr` to parse it back) - it took me a while to realise this!

Yeah it wasn't intended to be used for numbers really ...